s3/pkg/meta/redis/redis.go

```diff
@@ -0,0 +1,385 @@
+
+type RedisDatabase int
+
+func (r RedisDatabase) String() string {
+	return strconv.Itoa(int(r))
+}
+
+func (r RedisDatabase) InvalidQueue() string {
+	return InvalidQueueName + r.String()
+}
+
+const (
+	UserTable RedisDatabase = iota
+	BucketTable
+	ObjectTable
+	FileTable
+	ClusterTable
+)
+
+var MetadataTables = []RedisDatabase{UserTable, BucketTable, ObjectTable, ClusterTable}
+var DataTables = []RedisDatabase{FileTable}
+
+func Initialize(cfg *config.CommonConfig) {
+	redisClient = NewRedisCli()
+	redisClient.Init(cfg)
+}
+
+func Close() {
+	if redisClient != nil && redisClient.IsValid() {
+		err := redisClient.Close()
+		if err != nil {
+			helper.Logger.Printf(2, "Cannot close redis pool, err: %v", err)
+		}
+	}
+}
+
+func GetClient() (*RedisCli, error) {
+	return redisClient, nil
+}
+
+func HasRedisClient() bool {
+	if redisClient != nil && redisClient.IsValid() {
+		return true
+	}
+
+	return false
+}
+
+func Remove(table RedisDatabase, prefix, key string) (err error) {
+	c, err := GetClient()
+	if err != nil {
+		helper.Logger.Printf(2, "failed to get redis client, err: %v", err)
+		return err
+	}
+
+	// Use table.String() + hashkey as Redis key
+	_, err = c.Del(table.String() + prefix + helper.EscapeColon(key))
+	if err != nil {
+		helper.Logger.Printf(2, "failed to call redis del for (%s), err: %v", table.String()+prefix+helper.EscapeColon(key), err)
+		return err
+	}
+	helper.Logger.Printf(20, "Cmd: %s. Key: %s.", "DEL", table.String()+key)
+	return nil
+}
+
+func Set(table RedisDatabase, prefix, key string, value interface{}) (err error) {
+	c, err := GetClient()
+	if err != nil {
+		helper.Logger.Printf(2, "failed to get redis client, err: %v", err)
+		return err
+	}
+	encodedValue, err := helper.MsgPackMarshal(value)
+	if err != nil {
+		helper.Logger.Printf(2, "failed to make pack for(%s, %s, %v), err: %v", prefix, key, value, err)
+		return err
+	}
+
+	// Use table.String() + hashkey as Redis key. Set expire time to 30s.
+	r, err := c.Set(table.String()+prefix+helper.EscapeColon(key), encodedValue, 30000)
+	if err != nil {
+		helper.Logger.Printf(2, "failed to call redis Set(%s, %v), err: %v", table.String()+prefix+helper.EscapeColon(key), string(encodedValue), err)
+		return err
+	}
+	helper.Logger.Printf(20, "Cmd: %s. Key: %s. Value: %s. Reply: %s.", "SET", table.String()+key, string(encodedValue), r)
+	return nil
+}
+
+func Get(table RedisDatabase, prefix, key string) (value interface{}, err error) {
+	var encodedValue []byte
+	c, err := GetClient()
+	if err != nil {
+		helper.Logger.Printf(2, "failed to get redis client, err: %v", err)
+		return nil, err
+	}
+
+	// Use table.String() + hashkey as Redis key
+	encodedValue, err = c.Get(table.String() + prefix + helper.EscapeColon(key))
+	if err != nil {
+		helper.Logger.Printf(2, "failed to call redis Get(%s), err: %v", table.String()+prefix+helper.EscapeColon(key), err)
```

Comment:
this is a recurring pattern...it is not really a _failed to call redis_....we called some API and THAT failed and returned the error...can we reword our log ?